### PR TITLE
CSPL-570 Documentation for configuring indexer clusters and standalones with LM

### PR DESCRIPTION
**Pull request by akondur**
The JIRA [CSPL-570](https://jira.splunk.com/browse/CSPL-570) points out that the License Master(LM) logs are not forwarded to the Cluster Master(CM) which is configured as a license slave. The reason for this is that the LM has not setup the **outputs.conf** to send to the CM via indexer discovery. To enable the same on the LM, we need to introduce the **clusterMasterRef** in the LM Spec as show in the example below:

```
apiVersion: enterprise.splunk.com/v1beta1
kind: LicenseMaster
metadata:
  name: example
  finalizers:
  - enterprise.splunk.com/delete-pvc
spec:
  clusterMasterRef:
    name: example
  volumes:
    - name: licenses
      configMap:
        name: splunk-licenses
  licenseUrl: /mnt/licenses/enterprise.lic
```

The operator uses the **clusterMasterRef** to introduce the env variable **SPLUNK_CLUSTER_MASTER_URL** which is picked up ansible to setup indexer discovery on **outputs.conf**.

This PR edits the documentation with regards to connecting **IndexerClusters** and **Standalones** to **LicenseMaster**.

### CODE REVIEW

File: pkg/splunk/enterprise/configuration.go [hunk 1]
```diff
@@ -675,29 +675,33 @@ func updateSplunkPodTemplateWithConfig(client splcommon.ControllerClient, podTem
...
-		if err != nil {
-			scopedLog.Error(err, "Unable to get ClusterMaster")
-		}
 
-		if masterIdxCluster.Spec.LicenseMasterRef.Name != "" {
-			licenseMasterURL := GetSplunkServiceName(SplunkLicenseMaster, masterIdxCluster.Spec.LicenseMasterRef.Name, false)
-			if masterIdxCluster.Spec.LicenseMasterRef.Namespace != "" {
-				licenseMasterURL = splcommon.GetServiceFQDN(masterIdxCluster.Spec.LicenseMasterRef.Namespace, licenseMasterURL)
+		// Avoid for LM
+		if instanceType != SplunkLicenseMaster {
```
Comment (gaurav-splunk) [inline pkg/splunk/enterprise/configuration.go:680]: I am sorry but I have a trivial question here. If the instanceType is SplunkLicenseMaster, then how would it pick up the clusterMasterURL?

Comment (akondur) [reply pkg/splunk/enterprise/configuration.go:680]: Line 677. 

`		clusterMasterURL = GetSplunkServiceName(SplunkClusterMaster, spec.ClusterMasterRef.Name, false)
`

Followed by line 708

```
	if clusterMasterURL != "" {
		extraEnv = append(extraEnv, corev1.EnvVar{
			Name:  "SPLUNK_CLUSTER_MASTER_URL",
			Value: clusterMasterURL,
		})
	}
```

Comment (gaurav-splunk) [reply pkg/splunk/enterprise/configuration.go:680]: oh damn, I missed it completely

File: docs/Examples.md [hunk 2]
```diff
@@ -24,6 +26,26 @@ deployments.
...
+metadata:
+  name: example
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  licenseMasterRef:
+    name: example
+```
+
+## Connecting Indexer Clusters to use License Master
```
Comment (smohan-splunk) [inline docs/Examples.md:47]: This section seems incomplete? Is it needed?

File: docs/Examples.md [hunk 3]
```diff
@@ -24,6 +26,26 @@ deployments.
     - [Updating global kubernetes secret object](#updating-global-kubernetes-secret-object)
     - [Deleting global kubernetes secret object](#deleting-global-kubernetes-secret-object)
 
+## Connecting Splunk Enterprise CRs to use License Master
```
Comment (smohan-splunk) [inline docs/Examples.md:29]: Is this section necessary?

**Review by smohan-splunk — APPROVED** [review]

**Review by vebken-splunk — APPROVED** [review]

**Review by gaurav-splunk — APPROVED** [review]

**Pull request merged**